PyriteEditor/src/editor/views/Lights/widget_lights.h

```diff
@@ -169,6 +170,30 @@ namespace pye
 
 				ImGui::Checkbox("IsOn", &light.sourceLight->isOn);
 				ImGui::Separator();
+				ImGui::Checkbox("Cast dynamic shadows", (bool*)&light.sourceLight->shadowMode);
+				if (light.sourceLight->shadowMode == pyr::DynamicShadow && light.sourceLight->getType() != pyr::LightTypeID::Point)
+				{
+					ImGui::Text("Projection Parameters");
+					if (pyr::DirectionalLight* asDirectional = dynamic_cast<pyr::DirectionalLight*>(light.sourceLight))
```

Comment:
is the dynamic cast needed if you have `getType()`?

Reply:
yeah, cause the projection values are stored in the underlying light type. I need to dynamic cast the source baseLight... to be fair, the only reason the projection is not a parent member is because PointLights make no use of them, but then, i could still return 0. Might change this

Reply:
my point is that if you check the type using `getType()` you can static_cast instead of dynamic_cast

Reply:
right !